CMakeLists.txt

```diff
@@ -375,149 +393,74 @@ else()
-
-function(_make_distrib_subdir TARGET DIR)
-  set(NAME "_make_distrib_dir_${TARGET}")
-  add_custom_target("${NAME}" 
-                    COMMAND cmake -E make_directory "${DISTRIB_DIR}/${DIR}")
-  set_target_properties("${NAME}" PROPERTIES EXCLUDE_FROM_ALL TRUE)
-  add_dependencies("${TARGET}" "${NAME}" )
-endfunction()
-
-# Copy the file produced by TARGET into the distrib folder, in the subdir DIR
-function(add_to_distrib TARGET DIR)
-  set(options )
-  set(oneValueArgs )
-  set(multiValueArgs DEPS)
-  cmake_parse_arguments(args "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
-  if(TARGET "${TARGET}")
-    set(NAME "_distrib_${TARGET}")
-    get_target_property(TARGET_TYPE "${TARGET}" TYPE)
-    if("${TARGET_TYPE}" STREQUAL "EXECUTABLE" OR 
-       "${TARGET_TYPE}" STREQUAL "STATIC_LIBRARY" OR
-       "${TARGET_TYPE}" STREQUAL "SHARED_LIBRARY")
-      add_custom_target("${NAME}" 
-                         COMMAND cmake -E copy $<TARGET_FILE:${TARGET}> "${DISTRIB_DIR}/${DIR}"
-                         DEPENDS "${TARGET}")
-      _make_distrib_subdir("${NAME}" "${DIR}")
-    else()
-      message(FATAL_ERROR "TARGET ${TARGET} has type ${TARGET_TYPE}")
-    endif()
-    set_target_properties("${NAME}" PROPERTIES EXCLUDE_FROM_ALL TRUE)
-  else()
-    get_filename_component(FNAME "${TARGET}" NAME)
-    set(NAME "_distrib_${DIR}_${FNAME}")
-    string(REPLACE "/" "_" NAME "${NAME}")
-    string(REPLACE "\\" "_" NAME "${NAME}")
-    add_custom_target("${NAME}" 
-                       COMMAND cmake -E copy "${TARGET}" "${DISTRIB_DIR}/${DIR}"
-                       DEPENDS "${TARGET}")
-    _make_distrib_subdir("${NAME}" "${DIR}")
-  endif()
-  if(NOT "${args_DEPS}" STREQUAL "")
-    add_dependencies("${NAME}" ${args_DEPS})
-  endif()
-  add_dependencies(build_distrib_dir "${NAME}")
-endfunction()
-
-# ---- libHalide
-if("${HALIDE_LIBRARY_TYPE}" STREQUAL "SHARED")
-  add_to_distrib(Halide bin)
-else()
-  add_to_distrib(Halide lib)
-endif()
-
-# ---- Header files
-add_to_distrib("${CMAKE_BINARY_DIR}/include/Halide.h" include
-               DEPS HalideIncludes)
-add_to_distrib("${CMAKE_BINARY_DIR}/include/HalideBuffer.h" include)
-file(GLOB FILES "${CMAKE_BINARY_DIR}/include/HalideRuntime*.h")
-foreach(F ${FILES})
-  add_to_distrib(${F} include)
-endforeach()
-
-# ---- Tutorials
-foreach(EXT cpp h sh)
-  set(DIR "${CMAKE_SOURCE_DIR}/tutorial")
-  file(GLOB FILES "${DIR}/*.${EXT}")
-  foreach(F ${FILES})
-    add_to_distrib(${F} tutorial)
-  endforeach()
-endforeach()
-foreach(EXT gif jpg mp4)
-  set(DIR "${CMAKE_SOURCE_DIR}/tutorial/figures")
-  file(GLOB FILES "${DIR}/*.${EXT}")
-  foreach(F ${FILES})
-    add_to_distrib(${F} tutorial/figures)
-  endforeach()
-endforeach()
-foreach(EXT png)
-  set(DIR "${CMAKE_SOURCE_DIR}/tutorial/images")
-  file(GLOB FILES "${DIR}/*.${EXT}")
-  foreach(F ${FILES})
-    add_to_distrib(${F} tutorial/images)
-  endforeach()
-endforeach()
+# install
+
+install(FILES
```

Comment:
How does HalideRuntime*.h get included?

Reply:
Thanks for catching this. It was an unintentional omission. It's fixed now.